controllers/azurecluster_controller.go

```diff
@@ -106,7 +104,7 @@ func (r *AzureClusterReconciler) SetupWithManager(mgr ctrl.Manager, options cont
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=azureclusteridentities;azureclusteridentities/status,verbs=get;list;watch;create;update;patch;delete
 
 // Reconcile idempotently gets, creates, and updates a cluster.
-func (r *AzureClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
+func (r *AzureClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	ctx, cancel := context.WithTimeout(context.Background(), reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))
 	defer cancel()
```

Comment:
not sure what if this is the right thing to do here with context since it now has to be in the signature of the func, in capi they set it in main.go

@devigned @CecileRobertMichon

Reply:
We should also setup the signal handler in main.go. The signal handler listens to SIGTERM and SIGINT:
https://github.com/kubernetes-sigs/controller-runtime/blob/a763c9a36c6f18660799384c9765348942bda53a/pkg/manager/signals/signal.go#L30-L45

The intent of the loop timeout is to set a deterministic limit on the amount of time we will spend in a reconciliation loop. We should also keep the default loop timeout while passing in the outer context. 

```go
ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))`
```

Reply:
fixed for all of them